aries_cloudagent/registrar/base.py

```diff
@@ -0,0 +1,144 @@
+"""Base Class for DID registrars."""
+
+import logging
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import NamedTuple, Optional, Pattern
+
+from ..config.injection_context import InjectionContext
+from ..core.error import BaseError
+from ..core.profile import Profile
+
+
+class RegistrarError(BaseError):
+    """Base class for registrar exceptions."""
+
+
+class DIDNotFound(RegistrarError):
+    """Raised when DID is not found in verifiable data registry."""
+
+
+class DIDMethodNotSupported(RegistrarError):
+    """Raised when no registrar is registered for a given did method."""
+
+
+class RegistrarType(Enum):
+    """Registrar type declarations.
+
+    INTERNAL mode refers to when secrets are stored within this ACA-Py instance's
+    currently loaded secret storage system (Indy SDK Wallet, Askar, etc.)
+
+    HYBRID mode refers to when secrets are accessible to this ACA-Py instance
+    but not stored within ACA-Py's secret storage system.
+
+    EXTERNAL mode refers to when secrets are held outside of this ACA-Py
+    Instance altogether and are not accessible directly.
+    """
+
+    INTERNAL = "internal-secret-mode"
+    HYBRID = "hybrid-secret-mode"
+    EXTERNAL = "external-secret-mode"
+
+
+class IssueMetadata(NamedTuple):
+    """Issue Metadata."""
+
+    registrar_type: RegistrarType
+    registrar: str
+    retrieved_time: str
+    duration: int
+
+    def serialize(self) -> dict:
+        """Return serialized issue metadata."""
+        return {**self._asdict(), "registrar_type": self.registrar_type.value}
+
+
+class IssueResult:
+    """Issue Class to pack the DID Doc and the issue information."""
+
+    def __init__(self, did_document: dict, metadata: IssueMetadata):
+        """Initialize Issue.
+
+        Args:
+            did_doc: DID Document issued
+            registrar_metadata: Resolving details
+        """
+        self.did_document = did_document
+        self.metadata = metadata
+
+    def serialize(self) -> dict:
+        """Return serialized issue result."""
+        return {
+            "did_document": self.did_document,
+            "metadata": self.metadata.serialize(),
+        }
```

Comment:
Let's remove this for now; we can reintroduce these sorts of objects later if needed.